### PR TITLE
New CPD API

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: sonar-batch/src/main/java/org/sonar/batch/phases/PublishPhaseExecutor.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import org.sonar.batch.scan.filesystem.DefaultModuleFileSystem;
+import org.sonar.batch.scan.filesystem.FileSystemLogger;
+
+public final class PublishPhaseExecutor extends AbstractPhaseExecutor {
+
+  private final EventBus eventBus;
+  private final ReportPublisher reportPublisher;
+  private final CpdExecutor cpdExecutor;
+
+  public PublishPhaseExecutor(InitializersExecutor initializersExecutor, PostJobsExecutor postJobsExecutor, SensorsExecutor sensorsExecutor, SensorContext sensorContext,
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/phases/PublishPhaseExecutor.java]: MAJOR Constructor has 13 parameters, which is greater than 7 authorized. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00107)

File: sonar-batch/src/main/java/org/sonar/batch/phases/AbstractPhaseExecutor.java [hunk 2]
```diff
@@ -49,41 +42,29 @@
...
   private final QProfileVerifier profileVerifier;
   private final IssueExclusionsLoader issueExclusionsLoader;
-  private final IssuesReports issuesReport;
-  private final DefaultAnalysisMode analysisMode;
-  private final IssueTransition localIssueTracking;
-  private final IssueCallback issueCallback;
-  private final CpdExecutor cpdExecutor;
 
-  public PhaseExecutor(InitializersExecutor initializersExecutor, PostJobsExecutor postJobsExecutor, SensorsExecutor sensorsExecutor,
+  public AbstractPhaseExecutor(InitializersExecutor initializersExecutor, PostJobsExecutor postJobsExecutor, SensorsExecutor sensorsExecutor,
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/phases/AbstractPhaseExecutor.java]: MAJOR Constructor has 11 parameters, which is greater than 7 authorized. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00107)

File: sonar-batch/src/main/java/org/sonar/batch/phases/IssuesPhaseExecutor.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import org.sonar.batch.scan.report.IssuesReports;
+
+public final class IssuesPhaseExecutor extends AbstractPhaseExecutor {
+
+  private final EventBus eventBus;
+  private final IssuesReports issuesReport;
+  private final IssueTransition localIssueTracking;
+  private final IssueCallback issueCallback;
+
+  public IssuesPhaseExecutor(InitializersExecutor initializersExecutor, PostJobsExecutor postJobsExecutor, SensorsExecutor sensorsExecutor, SensorContext sensorContext,
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/phases/IssuesPhaseExecutor.java]: MAJOR Constructor has 14 parameters, which is greater than 7 authorized. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00107)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Comment by dbmeneses** [comment]
LGTM

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues:
* CRITICAL 1 critical
* MAJOR 3 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* CRITICAL [Either log or rethrow this exception.](https://github.com/SonarSource/sonarqube/blob/9cf13638be2e9d4a257183378a948554644079d5/sonar-batch/src/main/java/org/sonar/batch/cpd/CpdExecutor.java#L125) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1166)

**Pull request merged**